doc/pattern_tools/gen_pattern.py

```diff
@@ -111,12 +111,16 @@ def main():
     units = args.units
     square_size = args.square_size
     radius_rate = args.radius_rate
-    page_size = args.page_size
-    # page size dict (ISO standard, mm) for easy lookup. format - size: [width, height]
-    page_sizes = {"A0": [840, 1188], "A1": [594, 840], "A2": [420, 594], "A3": [297, 420], "A4": [210, 297],
-                  "A5": [148, 210]}
-    page_width = page_sizes[page_size.upper()][0]
-    page_height = page_sizes[page_size.upper()][1]
+    if 'page_width' and 'page_height' in args:
+        page_width = args.page_width
+        page_height = args.page_height
+    else:
+        page_size = args.page_size
+        # page size dict (ISO standard, mm) for easy lookup. format - size: [width, height]
+        page_sizes = {"A0": [840, 1188], "A1": [594, 840], "A2": [420, 594], "A3": [297, 420], "A4": [210, 297],
+                      "A5": [148, 210]}
+        page_width = page_sizes[page_size.upper()][0]
```

Comment:
Let make error messages user-friendly:
```
assert page_size.upper() in page_sizes, page_size.keys()
```

Reply:
@alalek That wasn't originally my code, but I did some tests and those .upper() calls are redundant. The choices list in argparse enforces a 1 to 1 match on the user entered string and the keys in page_sizes. I just went ahead and removed it altogether.